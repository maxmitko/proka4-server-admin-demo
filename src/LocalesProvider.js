import { locales } from 'locales'
import React from 'react';

export const LocalesContext = React.createContext();

export default props => <LocalesContext.Provider value={locales} children={props.children} />