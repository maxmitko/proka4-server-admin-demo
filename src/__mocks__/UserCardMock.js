export default {
    userCard: {
        data: [
            {
                id: 2,
                count: 11,
                price: 1500,
                debt: 0,
                isActive: 1,
                purchaseTime: '2019-01-28T16:36:53.000Z',
                user: {
                    id: 2,
                    username: 'Natalia',
                    fullname: 'Филатова Флорентина Леонидовна',
                    email: 'kras@mail.ru',
                    phone: '',
                    money: 670,
                    isActive: 1,
                    regDate: '2017-10-12T18:00:00.000Z',
                    lastAct: '2019-04-08T19:01:22.000Z'
                },
                service: {
                    id: 2,
                    title: 'Бодифлекс',
                    description: 'description 1',
                    isActive: 1,
                    myOrder: 1,
                    image: '0hpTp3jsx.jpeg'
                },
                visits: [
                    {
                        id: 445,
                        visitStatus: 1,
                        regDate: '2019-02-05T21:00:00.000Z',
                        regTime: '08:20:00'
                    },
                    {
                        id: 486,
                        visitStatus: 1,
                        regDate: '2019-02-20T21:00:00.000Z',
                        regTime: '08:20:00'
                    },
                    {
                        id: 489,
                        visitStatus: 1,
                        regDate: '2019-03-11T21:00:00.000Z',
                        regTime: '08:30:00'
                    },
                    {
                        id: 491,
                        visitStatus: 0,
                        regDate: '2019-03-30T21:00:00.000Z',
                        regTime: '18:00:00'
                    },
                    {
                        id: 495,
                        visitStatus: 1,
                        regDate: '2019-04-01T21:00:00.000Z',
                        regTime: '09:00:00'
                    },
                    {
                        id: 504,
                        visitStatus: 1,
                        regDate: '2019-04-01T21:00:00.000Z',
                        regTime: '08:15:00'
                    },
                    {
                        id: 506,
                        visitStatus: 1,
                        regDate: '2019-03-31T21:00:00.000Z',
                        regTime: '08:15:00'
                    }
                ]
            },
            {
                id: 66,
                count: 8,
                price: 1600,
                debt: 0,
                isActive: 1,
                purchaseTime: '2019-01-28T12:23:49.000Z',
                user: {
                    id: 2,
                    username: 'Natalia',
                    fullname: 'Филатова Флорентина Леонидовна',
                    email: 'kras@mail.ru',
                    phone: '',
                    money: 670,
                    isActive: 1,
                    regDate: '2017-10-12T18:00:00.000Z',
                    lastAct: '2019-04-08T19:01:22.000Z'
                },
                service: {
                    id: 5,
                    title: 'Тренинг "познай тело"',
                    description: 'description 2',
                    isActive: 1,
                    myOrder: 6,
                    image: 'dhA1V76uj.jpeg'
                },
                visits: [
                    {
                        id: 464,
                        visitStatus: 1,
                        regDate: '2019-02-11T21:00:00.000Z',
                        regTime: '08:20:00'
                    },
                    {
                        id: 476,
                        visitStatus: 1,
                        regDate: '2019-02-17T21:00:00.000Z',
                        regTime: '08:20:00'
                    }
                ]
            },
            {
                id: 92,
                count: 9,
                price: 420,
                debt: 0,
                isActive: 1,
                purchaseTime: '2019-02-20T03:52:30.000Z',
                user: {
                    id: 2,
                    username: 'Natalia',
                    fullname: 'Филатова Флорентина Леонидовна',
                    email: 'kras@mail.ru',
                    phone: '',
                    money: 670,
                    isActive: 1,
                    regDate: '2017-10-12T18:00:00.000Z',
                    lastAct: '2019-04-08T19:01:22.000Z'
                },
                service: {
                    id: 4,
                    title: 'Королевская осанка',
                    description: 'description 3',
                    isActive: 1,
                    myOrder: 4,
                    image: 'oZto6R1M5.jpeg'
                },
                visits: [
                    {
                        id: 500,
                        visitStatus: 1,
                        regDate: '2019-04-01T21:00:00.000Z',
                        regTime: '17:50:00'
                    }
                ]
            },
            {
                id: 93,
                count: 6,
                price: 700,
                debt: 0,
                isActive: 1,
                purchaseTime: '2019-02-20T03:52:32.000Z',
                user: {
                    id: 2,
                    username: 'Natalia',
                    fullname: 'Филатова Флорентина Леонидовна',
                    email: 'kras@mail.ru',
                    phone: '',
                    money: 670,
                    isActive: 1,
                    regDate: '2017-10-12T18:00:00.000Z',
                    lastAct: '2019-04-08T19:01:22.000Z'
                },
                service: {
                    id: 4,
                    title: 'Королевская осанка',
                    description: 'description 4',
                    isActive: 1,
                    myOrder: 4,
                    image: 'oZto6R1M5.jpeg'
                },
                visits: [
                    {
                        id: 501,
                        visitStatus: 1,
                        regDate: '2019-04-01T21:00:00.000Z',
                        regTime: '17:50:00'
                    }
                ]
            },
            {
                id: 94,
                count: 9,
                price: 800,
                debt: 0,
                isActive: 1,
                purchaseTime: '2019-02-20T03:52:35.000Z',
                user: {
                    id: 2,
                    username: 'Natalia',
                    fullname: 'Филатова Флорентина Леонидовна',
                    email: 'kras@mail.ru',
                    phone: '',
                    money: 670,
                    isActive: 1,
                    regDate: '2017-10-12T18:00:00.000Z',
                    lastAct: '2019-04-08T19:01:22.000Z'
                },
                service: {
                    id: 4,
                    title: 'Королевская осанка',
                    description: 'description 5',
                    isActive: 1,
                    myOrder: 4,
                    image: 'oZto6R1M5.jpeg'
                },
                visits: [
                    {
                        id: 502,
                        visitStatus: 1,
                        regDate: '2019-04-01T21:00:00.000Z',
                        regTime: '17:50:00'
                    }
                ]
            }
        ],
        totalCount: 0,
        options: {},
        error: false,
        loading: false,
        crud: {
            data: [],
            error: false,
            loading: false
        }
    }
}