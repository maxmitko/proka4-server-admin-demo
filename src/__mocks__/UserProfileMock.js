export default {
    formState: {
        header: "Редактировать",
        saveButton: "Сохранить",
        submitForm: () => { },
    },
    initialData: {
        email: "ovosh@mail.ru",
        fullname: "Моисеева Ульяна Дмитриевна",
        id: 5,
        isActive: 0,
        lastAct: "2019-04-03T19:18:33.000Z",
        money: 1200,
        phone: "+79829998733",
        regDate: "2019-02-18T22:10:14.000Z",
        username: "sweet",
    }
} 