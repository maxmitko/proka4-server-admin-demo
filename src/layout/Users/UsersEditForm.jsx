import CornerBadge from 'components/Badge/Corner'
import HeaderBadge from 'components/Badge/Header'
import Button from 'components/Button'
import Paper from 'components/Paper'
import React, { useCallback, useState } from 'react'
import { useMappedState } from 'redux-react-hook';
import styled from 'styled-components';

import UserCard from '../UserCard/UserCardTable'
import UserProfile from '../UserProfile/UserProfile'

const UsersEditForm = props => {

    const mapState = useCallback(state => ({ cardCount: state.userCard.data.length }), [])

    const { cardCount } = useMappedState(mapState);
    const { formState, initialData } = props;

    const [tabValue, setTabValue] = useState(0)

    return (
        <Wrapper>
            <HeaderBadge fullWidth>
                <TabsButtonWrapper>
                    <Button
                        active={tabValue === 0}
                        onClick={() => setTabValue(0)}
                        variant="text"
                        color="white" >Личные данные
                    </Button>

                    {!!initialData.id && <Button
                        active={tabValue === 1}
                        onClick={() => setTabValue(1)}
                        variant="text"
                        color="white" >Карта клиента
                        <CornerBadge
                            badgeContent={cardCount}
                            color={cardCount === 0 ? "warning" : "success"}
                        />
                    </Button>}
                </TabsButtonWrapper>
            </HeaderBadge>

            <TabContentWrapper>

                <Tab isVisible={tabValue === 0}>
                    <UserProfile formState={formState} initialData={initialData} />
                </Tab>

                <Tab isVisible={tabValue === 1}>
                    <UserCard userId={initialData.id} />
                </Tab>

            </TabContentWrapper>
        </Wrapper>
    )
}

export default UsersEditForm

const Wrapper = styled(Paper)`
    width: 750px;
`

export const Tab = styled.div`
    display: ${props => props.isVisible ? 'block' : 'none'};
`

export const TabContentWrapper = styled.div`
    margin: 20px 10px;
`

export const TabsButtonWrapper = styled.div`

    display: flex;
    align-items: center;

    > * {
        margin-right: 10px;
    }

    svg {
        margin-right: 8px;
        height: 21px;
    }
`