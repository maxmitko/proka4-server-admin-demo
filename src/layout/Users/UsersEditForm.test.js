import 'jest-dom/extend-expect'

import LocalesProvider from 'LocalesProvider'
import React from 'react';
import { cleanup, render, fireEvent } from 'react-testing-library'
import { createStore } from 'redux'
import { StoreContext } from 'redux-react-hook';
import { ThemeProvider } from 'styled-components'
import theme from 'theme/theme'

import UserCardMock from '../__mocks__/UserCardMock'
import UsersEditForm from './UsersEditForm';
import UserProfileMock from '../__mocks__/userProfileMock'
import produce from 'immer'

beforeAll(() => {
    const modalRoot = global.document.createElement('div');
    modalRoot.setAttribute('id', 'modal-root');
    const body = global.document.querySelector('body');
    body.appendChild(modalRoot);
});

afterEach(cleanup)

const mockStore = createStore((state, action) => UserCardMock);

const withHOC = ({ children }) => {
    return (
        <StoreContext.Provider value={mockStore}>
            <LocalesProvider>
                <ThemeProvider theme={theme}>
                    {children}
                </ThemeProvider >
            </LocalesProvider>
        </StoreContext.Provider>
    )
}

describe('react-testing UsersEditForm', () => {

    const customRender = (ui, options) => render(ui, { wrapper: withHOC, ...options })

    it('card button should not render', () => {
        const newProps = produce(UserProfileMock, draftState => {
            draftState.initialData.id = undefined
        })

        const { getByText } = customRender(<UsersEditForm {...newProps} />)

        expect(() => { getByText('Карта клиента') }).toThrow(/Unable to find an element/);
    });

    it('Button & CornerBadge should render correctly', () => {
        const { getByText } = customRender(<UsersEditForm {...UserProfileMock} />)

        expect(() => { getByText('Карта клиента') }).not.toThrow(/Unable to find an element/);
        expect(getByText('Карта клиента').querySelector('span').textContent).toBe("5");
    });

    it('default SNAPSHOT should render correctly', () => {
        const { asFragment } = customRender(<UsersEditForm {...UserProfileMock} />)
        expect(asFragment()).toMatchSnapshot();
    });

    it('classname should change after click', () => {
        const { getByText } = customRender(<UsersEditForm {...UserProfileMock} />)

        const profile = getByText('Личные данные').parentNode
        const card = getByText('Карта клиента').parentNode

        const temp = profile.className

        fireEvent.click(card)

        expect(temp).not.toEqual(profile.className);
    });
});