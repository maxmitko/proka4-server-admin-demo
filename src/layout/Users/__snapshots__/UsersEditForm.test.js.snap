// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-testing UsersEditForm default SNAPSHOT should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bZQynM sc-gipzik EllQY"
  >
    <span
      class="sc-bwzfXH bSXYSn"
    >
      <span
        class="sc-htpNat kudpsQ"
        color="default"
      >
        <div
          class="sc-bRBYWo bMxFny"
        >
          <button
            class="sc-bxivhb dzSBVV"
            color="white"
          >
            <div
              class="sc-EHOje fVHsnR"
            >
              Личные данные
            </div>
            <div
              class="rtr-root sc-ifAKCX jCCKIM"
              timeout="[object Object]"
            >
              <span />
            </div>
          </button>
          <button
            class="sc-bxivhb fMlgZn"
            color="white"
          >
            <div
              class="sc-EHOje fVHsnR"
            >
              Карта клиента
              <span
                class="sc-bdVaJa cFxWnN"
                color="success"
              >
                5
              </span>
            </div>
            <div
              class="rtr-root sc-ifAKCX jCCKIM"
              timeout="[object Object]"
            >
              <span />
            </div>
          </button>
        </div>
      </span>
    </span>
    <div
      class="sc-Rmtcm fnUFlH"
    >
      <div
        class="sc-csuQGl jGNKrR"
      >
        <form
          class="sc-jDwBTQ cDzFei"
        >
          <div
            class="sc-gPEVay bRSkdC"
          >
            <div
              class="sc-kAzzGY jkRSqS"
              color="default"
              label="Логин"
              value="sweet"
            >
              <label
                class="sc-cSHVUG cEERK"
              >
                Логин
              </label>
              <div
                class="sc-fjdhpX kbgEzK"
                color="default"
              >
                <input
                  class="sc-chPdSV iTTWNX"
                  value="sweet"
                />
                <div
                  class="sc-jzJRlG gkEvqg"
                  color="default"
                />
              </div>
            </div>
            <div
              class="sc-kAzzGY jkRSqS"
              color="default"
              label="ФИО"
              value="Моисеева Ульяна Дмитриевна"
            >
              <label
                class="sc-cSHVUG cEERK"
              >
                ФИО
              </label>
              <div
                class="sc-fjdhpX kbgEzK"
                color="default"
              >
                <input
                  class="sc-chPdSV iTTWNX"
                  value="Моисеева Ульяна Дмитриевна"
                />
                <div
                  class="sc-jzJRlG gkEvqg"
                  color="default"
                />
              </div>
            </div>
            <div
              class="sc-kAzzGY jkRSqS"
              color="default"
              label="Почта"
              value="ovosh@mail.ru"
            >
              <label
                class="sc-cSHVUG cEERK"
              >
                Почта
              </label>
              <div
                class="sc-fjdhpX kbgEzK"
                color="default"
              >
                <input
                  class="sc-chPdSV iTTWNX"
                  value="ovosh@mail.ru"
                />
                <div
                  class="sc-jzJRlG gkEvqg"
                  color="default"
                />
              </div>
            </div>
            <div
              class="sc-kAzzGY jkRSqS"
              color="default"
              label="Кошелек (руб)"
              value="1200"
            >
              <label
                class="sc-cSHVUG cEERK"
              >
                Кошелек (руб)
              </label>
              <div
                class="sc-fjdhpX kbgEzK"
                color="default"
              >
                <input
                  class="sc-chPdSV iTTWNX"
                  value="1200"
                />
                <div
                  class="sc-jzJRlG gkEvqg"
                  color="default"
                />
              </div>
            </div>
            <div
              class="sc-kAzzGY jkRSqS"
              color="default"
              label="Телефон"
              value="+79829998733"
            >
              <label
                class="sc-cSHVUG cEERK"
              >
                Телефон
              </label>
              <div
                class="sc-fjdhpX kbgEzK"
                color="default"
              >
                <input
                  class="sc-chPdSV iTTWNX"
                  value="+79829998733"
                />
                <div
                  class="sc-jzJRlG gkEvqg"
                  color="default"
                />
              </div>
            </div>
            <div
              class="sc-kAzzGY jkRSqS"
              color="default"
              label="Пароль"
              type="password"
              value=""
            >
              <label
                class="sc-cSHVUG cEERK"
              >
                Пароль
              </label>
              <div
                class="sc-fjdhpX kbgEzK"
                color="default"
              >
                <input
                  class="sc-chPdSV iTTWNX"
                  value=""
                />
                <div
                  class="sc-jzJRlG gkEvqg"
                  color="default"
                />
              </div>
            </div>
            <div
              class="sc-kAzzGY jkRSqS"
              color="default"
              label="Повторите пароль"
              type="password"
              value=""
            >
              <label
                class="sc-cSHVUG cEERK"
              >
                Повторите пароль
              </label>
              <div
                class="sc-fjdhpX kbgEzK"
                color="default"
              >
                <input
                  class="sc-chPdSV iTTWNX"
                  value=""
                />
                <div
                  class="sc-jzJRlG gkEvqg"
                  color="default"
                />
              </div>
            </div>
          </div>
          <div
            class="sc-iRbamj cUbavi"
          >
            <button
              class="sc-jlyJG emunYX sc-bxivhb hUEkAr"
              color="success"
              type="submit"
            >
              <div
                class="sc-EHOje fVHsnR"
              >
                Сохранить
              </div>
              <div
                class="rtr-root sc-ifAKCX jCCKIM"
                timeout="[object Object]"
              >
                <span />
              </div>
            </button>
          </div>
        </form>
      </div>
      <div
        class="sc-csuQGl ehqwrJ"
      >
        <div
          class="sc-jAaTju kRsMPV"
        >
          <div
            class="sc-cvbbAY eFzwdd"
          >
            <div
              class="sc-brqgnP hEgmVV"
            />
            <table
              class="sc-jWBwVP dUfZFW"
            >
              <thead>
                <tr>
                  <td
                    style="text-align: center;"
                  >
                    №
                  </td>
                  <td>
                    Курс
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    Стоимость
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    Долг
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    Квота
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    Остаток
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    Дата покупки
                  </td>
                  <td />
                </tr>
              </thead>
              <tbody>
                <tr
                  class="sc-eHgmQL UBTKg"
                >
                  <td
                    style="text-align: center;"
                  >
                    1
                  </td>
                  <td>
                    Бодифлекс
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    1500
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    0
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    11
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    5
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    28-01-2019
                  </td>
                  <td>
                    <div
                      class="sc-hSdWYo kdMaZE"
                    >
                      <button
                        class="sc-bxivhb fGTkgy"
                        color="success"
                      >
                        <div
                          class="sc-EHOje fVHsnR"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root-19"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                            />
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                          </svg>
                        </div>
                        <div
                          class="rtr-root sc-ifAKCX jCCKIM"
                          timeout="[object Object]"
                        >
                          <span />
                        </div>
                      </button>
                      <button
                        class="sc-bxivhb ciAmRZ"
                        color="danger"
                      >
                        <div
                          class="sc-EHOje fVHsnR"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root-19"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                            />
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                          </svg>
                        </div>
                        <div
                          class="rtr-root sc-ifAKCX jCCKIM"
                          timeout="[object Object]"
                        >
                          <span />
                        </div>
                      </button>
                    </div>
                  </td>
                </tr>
                <tr
                  class="sc-eHgmQL UBTKg"
                >
                  <td
                    style="text-align: center;"
                  >
                    2
                  </td>
                  <td>
                    Тренинг "познай тело"
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    1600
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    0
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    8
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    6
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    28-01-2019
                  </td>
                  <td>
                    <div
                      class="sc-hSdWYo kdMaZE"
                    >
                      <button
                        class="sc-bxivhb fGTkgy"
                        color="success"
                      >
                        <div
                          class="sc-EHOje fVHsnR"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root-19"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                            />
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                          </svg>
                        </div>
                        <div
                          class="rtr-root sc-ifAKCX jCCKIM"
                          timeout="[object Object]"
                        >
                          <span />
                        </div>
                      </button>
                      <button
                        class="sc-bxivhb ciAmRZ"
                        color="danger"
                      >
                        <div
                          class="sc-EHOje fVHsnR"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root-19"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                            />
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                          </svg>
                        </div>
                        <div
                          class="rtr-root sc-ifAKCX jCCKIM"
                          timeout="[object Object]"
                        >
                          <span />
                        </div>
                      </button>
                    </div>
                  </td>
                </tr>
                <tr
                  class="sc-eHgmQL UBTKg"
                >
                  <td
                    style="text-align: center;"
                  >
                    3
                  </td>
                  <td>
                    Королевская осанка
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    420
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    0
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    9
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    8
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    20-02-2019
                  </td>
                  <td>
                    <div
                      class="sc-hSdWYo kdMaZE"
                    >
                      <button
                        class="sc-bxivhb fGTkgy"
                        color="success"
                      >
                        <div
                          class="sc-EHOje fVHsnR"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root-19"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                            />
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                          </svg>
                        </div>
                        <div
                          class="rtr-root sc-ifAKCX jCCKIM"
                          timeout="[object Object]"
                        >
                          <span />
                        </div>
                      </button>
                      <button
                        class="sc-bxivhb ciAmRZ"
                        color="danger"
                      >
                        <div
                          class="sc-EHOje fVHsnR"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root-19"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                            />
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                          </svg>
                        </div>
                        <div
                          class="rtr-root sc-ifAKCX jCCKIM"
                          timeout="[object Object]"
                        >
                          <span />
                        </div>
                      </button>
                    </div>
                  </td>
                </tr>
                <tr
                  class="sc-eHgmQL UBTKg"
                >
                  <td
                    style="text-align: center;"
                  >
                    4
                  </td>
                  <td>
                    Королевская осанка
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    700
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    0
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    6
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    5
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    20-02-2019
                  </td>
                  <td>
                    <div
                      class="sc-hSdWYo kdMaZE"
                    >
                      <button
                        class="sc-bxivhb fGTkgy"
                        color="success"
                      >
                        <div
                          class="sc-EHOje fVHsnR"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root-19"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                            />
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                          </svg>
                        </div>
                        <div
                          class="rtr-root sc-ifAKCX jCCKIM"
                          timeout="[object Object]"
                        >
                          <span />
                        </div>
                      </button>
                      <button
                        class="sc-bxivhb ciAmRZ"
                        color="danger"
                      >
                        <div
                          class="sc-EHOje fVHsnR"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root-19"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                            />
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                          </svg>
                        </div>
                        <div
                          class="rtr-root sc-ifAKCX jCCKIM"
                          timeout="[object Object]"
                        >
                          <span />
                        </div>
                      </button>
                    </div>
                  </td>
                </tr>
                <tr
                  class="sc-eHgmQL UBTKg"
                >
                  <td
                    style="text-align: center;"
                  >
                    5
                  </td>
                  <td>
                    Королевская осанка
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    800
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    0
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    9
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    8
                  </td>
                  <td
                    style="text-align: center;"
                  >
                    20-02-2019
                  </td>
                  <td>
                    <div
                      class="sc-hSdWYo kdMaZE"
                    >
                      <button
                        class="sc-bxivhb fGTkgy"
                        color="success"
                      >
                        <div
                          class="sc-EHOje fVHsnR"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root-19"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                            />
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                          </svg>
                        </div>
                        <div
                          class="rtr-root sc-ifAKCX jCCKIM"
                          timeout="[object Object]"
                        >
                          <span />
                        </div>
                      </button>
                      <button
                        class="sc-bxivhb ciAmRZ"
                        color="danger"
                      >
                        <div
                          class="sc-EHOje fVHsnR"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root-19"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                            />
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                          </svg>
                        </div>
                        <div
                          class="rtr-root sc-ifAKCX jCCKIM"
                          timeout="[object Object]"
                        >
                          <span />
                        </div>
                      </button>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="sc-hzDkRC deNlEF"
        >
          <div
            class="sc-kAzzGY jkRSqS"
            color="default"
            data=""
            placeholder="Выбирите абонимент"
            value=""
          >
            <label
              class="sc-cSHVUG dWFnGS"
            />
            <div
              class="sc-fjdhpX gcKgED"
              color="default"
            >
              <div
                class="sc-kgoBCf dOcyB"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-19"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M20.57 14.86L22 13.43 20.57 12 17 15.57 8.43 7 12 3.43 10.57 2 9.14 3.43 7.71 2 5.57 4.14 4.14 2.71 2.71 4.14l1.43 1.43L2 7.71l1.43 1.43L2 10.57 3.43 12 7 8.43 15.57 17 12 20.57 13.43 22l1.43-1.43L16.29 22l2.14-2.14 1.43 1.43 1.43-1.43-1.43-1.43L22 16.29z"
                  />
                </svg>
              </div>
              <input
                class="sc-chPdSV iTTWNX"
                placeholder="Выбирите абонимент"
                value=""
              />
              <div
                class="sc-kGXeez jHAfAq"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-19"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </div>
              <div
                class="sc-jzJRlG dNgqWR"
                color="default"
              />
            </div>
          </div>
          <button
            class="sc-bxivhb cZkzNO"
            color="default"
            disabled=""
            type="submit"
          >
            <div
              class="sc-EHOje fVHsnR"
            >
              Добавить
            </div>
            <div
              class="rtr-root sc-ifAKCX jCCKIM"
              timeout="[object Object]"
            >
              <span />
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
